src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1256,6 +1310,37 @@ public bool ValidateTransaction(Transaction transaction, bool checkSignature = f
             }
         }
 
+        /// <summary>
+        /// Determines if a mempool error would NOT be recoverable by waiting or rebuilding the transaction.
+        /// </summary>
+        /// <param name="mempoolError">The error to evailuate.</param>
+        /// <returns><c>true</c> if its irrecoverble or <c>false</c> otherwise.</returns>
+        public static bool IsMempoolErrorRecoverable(MempoolError mempoolError)
```

Comment:
Would be nice to have a unit test for this.

Reply:
Done.